src/webui/webapplication.cpp

```diff
@@ -538,453 +630,911 @@ void WebApplication::action_command_addTrackers()
 {
-    CHECK_URI(0);
-    CHECK_PARAMETERS("category");
+    checkParams({"category"});
 
-    QString category = request().posts["category"].trimmed();
+    QString category = params()["category"].trimmed();
 
     if (!BitTorrent::Session::isValidCategoryName(category) && !category.isEmpty()) {
-        status(400, tr("Incorrect category name"));
+        status(409, "Conflict");
+        print(tr("Incorrect category name"), Http::CONTENT_TYPE_TXT);
         return;
     }
 
     BitTorrent::Session::instance()->addCategory(category);
 }
 
-void WebApplication::action_command_removeCategories()
+void WebApplication::categoriesRemoveAction()
 {
-    CHECK_URI(0);
-    CHECK_PARAMETERS("categories");
+    checkParams({"categories"});
 
-    QStringList categories = request().posts["categories"].split('\n');
+    QStringList categories = params()["categories"].split('\n');
     foreach (const QString &category, categories)
         BitTorrent::Session::instance()->removeCategory(category);
 }
 
-void WebApplication::action_command_getSavePath()
+void WebApplication::appDefaultSavePathAction()
 {
-    CHECK_URI(0);
     print(BitTorrent::Session::instance()->defaultSavePath());
 }
 
-bool WebApplication::isPublicScope()
+void WebApplication::rssAddFolderAction()
+{
+    checkParams({"path"});
+
+    const QString path = params()["path"].trimmed();
+    QString error;
+    if (!RSS::Session::instance()->addFolder(path, &error)) {
+        status(409, "Conflict");
+        print(error, Http::CONTENT_TYPE_TXT);
+    }
+}
+
+void WebApplication::rssAddFeedAction()
+{
+    checkParams({"url", "path"});
+
+    const QString url = params()["url"].trimmed();
+    const QString path = params()["path"].trimmed();
+    QString error;
+    if (!RSS::Session::instance()->addFeed(url, (path.isEmpty() ? url : path), &error)) {
+        status(409, "Conflict");
+        print(error, Http::CONTENT_TYPE_TXT);
+    }
+}
+
+void WebApplication::rssRemoveItemAction()
+{
+    checkParams({"path"});
+
+    const QString path = params()["path"].trimmed();
+    QString error;
+    if (!RSS::Session::instance()->removeItem(path, &error)) {
+        status(409, "Conflict");
+        print(error, Http::CONTENT_TYPE_TXT);
+    }
+}
+
+void WebApplication::rssMoveItemAction()
+{
+    checkParams({"itemPath", "destPath"});
+
+    const QString itemPath = params()["itemPath"].trimmed();
+    const QString destPath = params()["destPath"].trimmed();
+    QString error;
+    if (!RSS::Session::instance()->moveItem(itemPath, destPath, &error)) {
+        status(409, "Conflict");
+        print(error, Http::CONTENT_TYPE_TXT);
+    }
+}
+
+void WebApplication::rssItemsAction()
+{
+    checkParams({}, {"withData"});
+
+    const bool withData {parseBool(params()["withData"], false)};
+
+    const auto jsonVal {RSS::Session::instance()->rootFolder()->toJsonValue(withData)};
+    print(QJsonDocument(jsonVal.toObject()).toJson(), Http::CONTENT_TYPE_JSON);
```

Comment:
On compile I get an error

```
webui/webapplication.cpp: In member function ‘void WebApplication::rssItemsAction()’:
webui/webapplication.cpp:1013:33: error: ‘const class std::initializer_list<const QJsonValue>’ has no member named ‘toObject’
     print(QJsonDocument(jsonVal.toObject()).toJson(), Http::CONTENT_TYPE_JSON);
                                 ^
webui/webapplication.cpp: In member function ‘void WebApplication::rssSetRuleAction()’:
webui/webapplication.cpp:1024:104: error: no matching function for call to ‘RSS::AutoDownloadRule::fromJsonObject(const std::initializer_list<const QJsonObject>&, const QString&)’
     RSS::AutoDownloader::instance()->insertRule(RSS::AutoDownloadRule::fromJsonObject(jsonObj, ruleName));
                                                                                                        ^
webui/webapplication.cpp:1024:104: note: candidate is:
In file included from webui/webapplication.cpp:61:0:
./base/rss/rss_autodownloadrule.h:86:33: note: static RSS::AutoDownloadRule RSS::AutoDownloadRule::fromJsonObject(const QJsonObject&, const QString&)
         static AutoDownloadRule fromJsonObject(const QJsonObject &jsonObj, const QString &name = "");
                                 ^
./base/rss/rss_autodownloadrule.h:86:33: note:   no known conversion for argument 1 from ‘const std::initializer_list<const QJsonObject>’ to ‘const QJsonObject&’
```

Reply:
@glassez Still getting this error with the latest commits

Reply:
Not sure with the last one... 
Travis compiles it without errors now, so either you have outdated code version, or you're not doing a clean build.